### PR TITLE
Fix #5248 - Incorrect math in water coil warning messages

### LINKED ISSUE #5248: Incorrect math in water coil warning messages
Math used to trip warnings for water coils is incorrect.

These warning tests should use \* AirDensity
if (  AirMassFlow > ( 5.0 \* WaterCoil( CoilNum ).MinAirFlowArea / AirDensity )...
kg/s = m/s \* m2 \* kg/m3

Also, math in several places within these 2 warnings need correcting. No defect file is available since this was found accidentally using the wrong idd.

Line 2414 of WaterCoils:

```
//Warning and error messages for large flow rates for the given user input geometry
AirDensity = PsyRhoAirFnPbTdbW( OutBaroPress, TempAirIn, InletAirHumRat, RoutineName );
if ( AirMassFlow > ( 5.0 * WaterCoil( CoilNum ).MinAirFlowArea / AirDensity ) && CoilWarningOnceFlag( CoilNum ) ) {
    ShowWarningError( "Coil:Cooling:Water:DetailedGeometry in Coil =" + WaterCoil( CoilNum ).Name );
    ShowContinueError( "Air Flow Rate Velocity has greatly exceeded upper design guidelines of ~2.5 m/s" );
    ShowContinueError( "Air MassFlowRate[kg/s]=" + TrimSigDigits( AirMassFlow, 6 ) );
    AirVelocity = AirMassFlow * AirDensity / WaterCoil( CoilNum ).MinAirFlowArea;
    ShowContinueError( "Air Face Velocity[m/s]=" + TrimSigDigits( AirVelocity, 6 ) );
    ShowContinueError( "Approximate MassFlowRate limit for Face Area[kg/s]=" + TrimSigDigits( 2.5 * WaterCoil( CoilNum ).MinAirFlowArea / AirDensity, 6 ) );
    ShowContinueError( "Coil:Cooling:Water:DetailedGeometry could be resized/autosized to handle capacity" );
    CoilWarningOnceFlag( CoilNum ) = false;
} else if ( AirMassFlow > ( 44.7 * WaterCoil( CoilNum ).MinAirFlowArea / AirDensity ) ) {
    ShowSevereError( "Coil:Cooling:Water:DetailedGeometry in Coil =" + WaterCoil( CoilNum ).Name );
    ShowContinueError( "Air Flow Rate Velocity is > 100MPH (44.7m/s) and simulation cannot continue" );
    ShowContinueError( "Air Mass Flow Rate[kg/s]=" + TrimSigDigits( AirMassFlow, 6 ) );
    AirVelocity = AirMassFlow * AirDensity / WaterCoil( CoilNum ).MinAirFlowArea;
    ShowContinueError( "Air Face Velocity[m/s]=" + TrimSigDigits( AirVelocity, 6 ) );
    ShowContinueError( "Approximate MassFlowRate limit for Face Area[kg/s]=" + TrimSigDigits( 2.5 * WaterCoil( CoilNum ).MinAirFlowArea / AirDensity, 6 ) );
    ShowFatalError( "Coil:Cooling:Water:DetailedGeometry needs to be resized/autosized to handle capacity" );
}
```

densityunits

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
- Fixes #5248
- Incorrect math in water coil warning messages

No defect files existed, but the unit test captures the problem and tests the new behavior.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/WaterCoils.cc [hunk 1]
```diff
@@ -3105,26 +3105,29 @@ namespace WaterCoils {
...
             ShowContinueError(state, "Air Flow Rate Velocity is > 100MPH (44.7m/s) and simulation cannot continue");
             ShowContinueError(state, format("Air Mass Flow Rate[kg/s]={:.6T}", AirMassFlow));
-            AirVelocity = AirMassFlow * AirDensity / state.dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea;
+            AirVelocity = AirMassFlow / (state.dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea * AirDensity);
             ShowContinueError(state, format("Air Face Velocity[m/s]={:.6T}", AirVelocity));
             ShowContinueError(state,
-                              format("Approximate MassFlowRate limit for Face Area[kg/s]={:.6T}",
-                                     2.5 * state.dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea / AirDensity));
+                              format("Approximate Mass Flow Rate limit for Face Area[kg/s]={:.6T}",
+                                     44.7 * state.dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea * AirDensity));
```
Comment (jmarrec) [inline src/EnergyPlus/WaterCoils.cc]: Adjust the math and the text of the warning/fatal too.

File: tst/EnergyPlus/unit/WaterCoils.unit.cc [hunk 2]
```diff
@@ -1054,6 +1055,189 @@ TEST_F(WaterCoilsTest, CoilCoolingWaterDetailedSizing)
...
+        format("   **   ~~~   ** Air Face Velocity[m/s]={:.6T}",
+               AirMassFlow / (state->dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea * AirDensity)),
+        format("   **   ~~~   ** Approximate Mass Flow Rate limit for Face Area[kg/s]={:.6T}",
+               2.5 * state->dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea * AirDensity),
+        "   **   ~~~   ** Coil:Cooling:Water:DetailedGeometry could be resized/autosized to handle capacity",
+    });
+
+    CalcDetailFlatFinCoolingCoil(*state, CoilNum, 2, 2, 1);
+
+    compare_err_stream(expected_error, true);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/WaterCoils.unit.cc]: Tests the case where it's above 5.0 m/s

File: tst/EnergyPlus/unit/WaterCoils.unit.cc [hunk 3]
```diff
@@ -1054,6 +1055,189 @@ TEST_F(WaterCoilsTest, CoilCoolingWaterDetailedSizing)
...
+                44.7 * state->dataWaterCoils->WaterCoil(CoilNum).MinAirFlowArea * AirDensity),
+        "   **  Fatal  ** Coil:Cooling:Water:DetailedGeometry needs to be resized/autosized to handle capacity",
+        "   ...Summary of Errors that led to program termination:",
+        "   ..... Reference severe error count=1",
+        "   ..... Last severe error=Coil:Cooling:Water:DetailedGeometry in Coil =Test Detailed Water Cooling Coil",
+    });
+
+    EXPECT_ANY_THROW(CalcDetailFlatFinCoolingCoil(*state, CoilNum, 2, 2, 1));
+
+    compare_err_stream(expected_fatal_error, true);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/WaterCoils.unit.cc]: Tests the case where it's above 44.7 m/s (100 MPH) where it fatals out.

**Comment by jmarrec** [comment]
Two tests have expected ERR diffs:

## regression.FreeCoolingChiller

With the correct math, this test no longer triggers the > 5.0 m/s warning.

```diff
--- 
+++ 
@@ -1,11 +1,5 @@
    ** Warning ** GetSurfaceData: World Coordinate System selected.  Any non-zero Building/Zone North Axes or non-zero Zone Origins are ignored.
    **   ~~~   ** These may be used in daylighting reference point coordinate calculations but not in normal geometry inputs.
-   ** Warning ** Coil:Cooling:Water:DetailedGeometry in Coil =DETAILED COOLING COIL
-   **   ~~~   ** Air Flow Rate Velocity has greatly exceeded upper design guidelines of ~2.5 m/s
-   **   ~~~   ** Air MassFlowRate[kg/s]=29.397116
-   **   ~~~   ** Air Face Velocity[m/s]=6.041382
-   **   ~~~   ** Approximate MassFlowRate limit for Face Area[kg/s]=12.164895
-   **   ~~~   ** Coil:Cooling:Water:DetailedGeometry could be resized/autosized to handle capacity
    ************* Testing Individual Branch Integrity
    ************* All Branches passed integrity testing
    ************* Testing Individual Supply Air Path Integrity
```

## regression.Plenum

The warning is changed to reflect the new math:

```diff
--- 
+++ 
@@ -1,8 +1,8 @@
    ** Warning ** Coil:Cooling:Water:DetailedGeometry in Coil =DETAILED COOLING COIL
    **   ~~~   ** Air Flow Rate Velocity has greatly exceeded upper design guidelines of ~2.5 m/s
-   **   ~~~   ** Air MassFlowRate[kg/s]=5.056304
-   **   ~~~   ** Air Face Velocity[m/s]=7.050460
-   **   ~~~   ** Approximate MassFlowRate limit for Face Area[kg/s]=1.792898
+   **   ~~~   ** Air Mass Flow Rate[kg/s]=5.056304
+   **   ~~~   ** Air Face Velocity[m/s]=5.518599
+   **   ~~~   ** Approximate Mass Flow Rate limit for Face Area[kg/s]=2.290574
    **   ~~~   ** Coil:Cooling:Water:DetailedGeometry could be resized/autosized to handle capacity
    ************* Testing Individual Branch Integrity
    ************* All Branches passed integrity testing
```

**Comment by mitchute** [comment]
The changes look good here. I expect this can go in shortly after CI reports back.

**Comment by mitchute** [comment]
Good here with expected diffs. Merging.

**Pull request merged**